graph/db/benchmark_test.go

```diff
@@ -0,0 +1,703 @@
+		"channeldb",
+	)
+	require.NoError(t, err)
+
+	return kvStore
+}
+
+// connectKVDBSqlite creates a V1Store instance backed by a kvdb-sqlite
+// database for testing purposes.
+func connectKVDBSqlite(t testing.TB, dbPath, fileName string) V1Store {
+	return newKVStore(t, kvdbSqlite(t, dbPath, fileName))
+}
+
+// connectBBoltDB creates a new BBolt database connection for testing.
+func connectBBoltDB(t testing.TB, dbPath, fileName string) V1Store {
+	cfg := &kvdb.BoltBackendConfig{
+		DBPath:            dbPath,
+		DBFileName:        fileName,
+		NoFreelistSync:    true,
+		AutoCompact:       false,
+		AutoCompactMinAge: kvdb.DefaultBoltAutoCompactMinAge,
+		DBTimeout:         kvdb.DefaultDBTimeout,
+	}
+
+	kvStore, err := kvdb.GetBoltBackend(cfg)
+	require.NoError(t, err)
+
+	return newKVStore(t, kvStore)
+}
+
+// newKVStore creates a new KVStore instance for testing using a provided
+// kvdb.Backend instance.
+func newKVStore(t testing.TB, backend kvdb.Backend) V1Store {
+	store, err := NewKVStore(backend, testStoreOptions...)
+	require.NoError(t, err)
+
+	return store
+}
+
+// newSQLExecutor creates a new BatchedSQLQueries instance for testing using a
+// provided sqldb.DB instance.
+func newSQLExecutor(t testing.TB, db sqldb.DB) BatchedSQLQueries {
+	err := db.ApplyAllMigrations(
+		context.Background(), sqldb.GetMigrations(),
+	)
+	require.NoError(t, err)
+
+	return sqldb.NewTransactionExecutor(
+		db.GetBaseDB(), func(tx *sql.Tx) SQLQueries {
+			return db.GetBaseDB().WithTx(tx)
+		},
+	)
+}
+
+// newSQLStore creates a new SQLStore instance for testing using a provided
+// sqldb.DB instance.
+func newSQLStore(t testing.TB, db BatchedSQLQueries) V1Store {
+	store, err := NewSQLStore(
+		&SQLStoreConfig{
+			ChainHash:     dbTestChain,
+			PaginationCfg: testSQLPaginationCfg,
+		},
+		db, testStoreOptions...,
+	)
+	require.NoError(t, err)
+
+	return store
+}
+
+// TestPopulateDBs is a helper test that can be used to populate various local
+// graph DBs from some source graph DB. This can then be used to run the
+// various benchmark tests against the same graph data.
+//
+// TODO(elle): this test reveals that the batching logic we use might be
+// problematic for postgres backends. This needs some investigation & it might
+// make sense to only use LazyAdd for sqlite backends & it may make sense to
+// also add a maximum batch size to avoid grouping too many updates at once.
+// Observations:
+//   - the LazyAdd options need to be turned off for channel/policy update calls
+//     for both the native SQL postgres & kvdb postgres backend for this test to
+//     succeed.
+//   - The LazyAdd option must be added for the sqlite backends, else it takes
+//     very long to sync the graph.
+func TestPopulateDBs(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+
+	// NOTE: uncomment the line below to run this test locally.
+	t.Skipf("Skipping local helper test")
+
+	// Set your desired source database.
+	sourceDB := kvdbSqliteConn
```

Comment:
It should be pretty obvious from all the constants at the start of the file and from what the benchmark test actually does.
But perhaps a comment like this will help someone in the future:

```
	// NOTE: uncomment the line below to run this test locally, then provide
	// the desired source database (and make sure the destination Postgres
	// databases exist and are running).
	t.Skipf("Skipping local helper test")

	// Set your desired source database here. For kvdbSqliteConn, a file
	// called testdata/kvdb/channel.sqlite must exist and be populated with
	// a KVDB based channel graph.
	sourceDB := kvdbSqliteConn
```